bufferserver/src/main/java/com/datatorrent/bufferserver/server/Server.java

```diff
@@ -530,16 +547,13 @@ public int readSize()
 
   class Subscriber extends AbstractLengthPrependerClient
   {
-    private final String type;
-    private final int mask;
-    private final int[] partitions;
+    private LogicalNode ln;
 
-    Subscriber(String type, int mask, int[] partitions, int bufferSize)
+    Subscriber(LogicalNode ln, int bufferSize)
```

Comment:
Not, when it is created. It is possible that `ln` will be set to null after Subscriber teardown.